test/e2eprovider/vault/vault.go

```diff
@@ -0,0 +1,177 @@
+package e2eprovider
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+// E2eVault is a fake Vault for end-to-end testing.
+type E2eVault struct {
+	KubeClient         *kubernetes.Clientset
+	VaultNamespaceName string
+}
+
+// Vault is the interface for interacting with fake Vault.
+type Vault interface {
+	// GetSecret returns the value of the given secret.
+	GetSecret(name string) (string, string, error)
+}
+
+// NewVault creates a new fake Vault.
+func NewVault() (Vault, error) {
+	kubeClient := getClientSet()
+	vaultNamespaceName, err := populateVault(kubeClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return &E2eVault{
+		KubeClient:         kubeClient,
+		VaultNamespaceName: vaultNamespaceName,
+	}, nil
+}
+
+// GetSecret returns the value of the given secret.
+func (v *E2eVault) GetSecret(name string) (string, string, error) {
+	secret, err := v.KubeClient.CoreV1().Secrets(v.VaultNamespaceName).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return "", "", err
+	}
+
+	return string(secret.Data[name]), secret.GetLabels()["version"], nil
+}
+
+// populateVault populates the fake Vault with default secrets.
+func populateVault(kubeClient *kubernetes.Clientset) (string, error) {
```

Comment:
IMO, this additional logic to create a backend with Kubernetes secret introduces an additional error path in the e2e which don't want to have. What if we just had the `e2e-provider` implementation trimmed down to return static values to perform a deterministic value check in the bats suite? That reduces the scope of failures to be specific for driver during e2e runs which is the intent. I like how we have are defining the interface to show how this can be useful for actual provider implementation but the `GetSecret` can just return a simple value?

cc @tam7t

Reply:
switched back to returning `static values` return rather than `K8s secrets as vault`